src/adapters/curve/products/pool/curvePoolAdapter.ts

```diff
@@ -38,54 +51,83 @@ export class CurvePoolAdapter
 {
   productId = 'pool'
 
-  /**
-   * Update me.
-   * Add your protocol details
-   */
   getProtocolDetails(): ProtocolDetails {
     return {
       protocolId: this.protocolId,
       name: 'Curve',
       description: 'Curve pool adapter',
-      siteUrl: 'https:',
-      iconUrl: 'https://',
+      siteUrl: 'https://curve.fi/',
+      iconUrl:
+        'https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/crv.svg',
       positionType: PositionType.Supply,
       chainId: this.chainId,
       productId: this.productId,
     }
   }
 
-  /**
-   * Update me.
-   * Add logic to build protocol token metadata
-   * For context see dashboard example ./dashboard.png
-   * We need protocol token names, decimals, and also linked underlying tokens
-   */
   @CacheToFile({ fileKey: 'protocol-token' })
   async buildMetadata() {
-    return {} as CurvePoolAdapterMetadata
+    const metaRegistryContract = MetaRegistry__factory.connect(
+      CURVE_META_REGISTRY_CONTRACT,
+      this.provider,
+    )
+
+    const metadataObject: CurvePoolAdapterMetadata = {}
+    const poolCount = Number(await metaRegistryContract.pool_count())
+    const poolDataPromises = Array.from({ length: poolCount }, (_, i) =>
+      this.getPoolData(i),
+    )
+    const results = await Promise.all(poolDataPromises)
+
+    filterMapSync(results, (token) => {
+      if (!token) {
+        return undefined
+      }
+
+      metadataObject[token.protocolToken.address] = token
+    })
+
+    return metadataObject
   }
 
-  /**
-   * Update me.
-   * Below implementation might fit your metadata if not update it.
-   */
   async getProtocolTokens(): Promise<Erc20Metadata[]> {
     return Object.values(await this.buildMetadata()).map(
       ({ protocolToken }) => protocolToken,
     )
   }
 
-  /**
-   * Update me.
-   * Add logic to turn the LP token balance into the correct underlying token(s) balance
-   * For context see dashboard example ./dashboard.png
-   */
   protected async getUnderlyingTokenBalances(
-    _protocolTokenBalance: TokenBalance,
-    _blockNumber?: number,
+    protocolTokenBalance: TokenBalance,
+    blockNumber: number,
   ): Promise<Underlying[]> {
-    throw new NotImplementedError()
+    const { protocolToken } = await this.fetchPoolMetadata(
+      protocolTokenBalance.address,
+    )
```

Comment:
nit: You can use `this.fetchProtocolTokenMetadata`, which is already implemented.